### PR TITLE
Remove x87 floating point comparisons

### PR DESCRIPTION
**Pull request by 0xdaryl**
Miscellaneous initial commits to remove the x87 implementation of floating point comparisons.

Issue #946

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins build xlinux,win,osx,x32linux

**Comment by 0xdaryl** [comment]
@BradleyWood : please review.

File: compiler/x/codegen/OMRX86Instruction.cpp
Comment (dsouzai) [inline compiler/x/codegen/OMRX86Instruction.cpp:3076]: Maybe I'm misunderstanding, but FCOMI instructions are x87 instructions right? If the compiler can't use the FCOMI instructions, then shouldn't we be using the falsy expression (ie `TR::InstOpCode::JNE4`)? Same for all the changes below.

**Review by dsouzai — COMMENTED** [review]
Looks ok to me; just waiting on the answer to the one review question before approval/merge.

File: compiler/x/codegen/OMRX86Instruction.cpp
Comment (dsouzai) [reply compiler/x/codegen/OMRX86Instruction.cpp:3076]: Ah I see, I got confused and thought the FCOMI instructions were x87 because queries around them were also removed, but I guess these were removed because there's no point querying the default behaviour.

**Review by dsouzai — APPROVED** [review]

**Pull request merged**
